pkg/api/object.go

```diff
@@ -49,3 +52,31 @@ func (h *Handler) RemoveObjectStore(w http.ResponseWriter, r *http.Request) {
 	logger.Debugf("started async deletion of the object store")
 	w.WriteHeader(http.StatusAccepted)
 }
+
+// Gets connection information to the object store in this cluster.
+// GET
+// /objectstore/connectioninfo
+func (h *Handler) GetObjectStoreConnectionInfo(w http.ResponseWriter, r *http.Request) {
+	// TODO: auth is extremely important here because we are returning RGW credentials
+
+	// TODO: support other types of connection info to object store (such as swift)
+
+	accessID, secretKey, err := rgw.GetBuiltinUserAccessInfo(h.context.EtcdClient)
+	if err != nil {
+		logger.Errorf("failed to retrieve builtin user access info: %+v", err)
+		if util.IsEtcdKeyNotFound(err) {
+			w.WriteHeader(http.StatusNotFound)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	s3Info := model.ObjectStoreS3Info{
+		Host:      fmt.Sprintf("%s:%d", rgw.DNSName, rgw.RGWPort),
```

Comment:
no, the DNS name, even when put into /etc/hosts, did not work for me with s3cmd.  It would be useful to return both I suppose, I will give that a try.